src/collection.rs

```diff
@@ -311,6 +311,12 @@ impl<T: Id<T>> CollectionWithId<T> {
         }
     }
 
+    /// Returns a reference to the object corresponding to the
+    /// identifier.
```

Comment:
this comment is wrong.
@TeXitoi can you help me to find a description and a good example please?

Reply:
I think you should take inspiration if index_mut. Thus, something like 

> Access to a mutable reference of the object corresponding to the identifier.
>
> The `drop` of the proxy object panic if the identifier is modified to an identifier already on the collection.

and an example close to index_mut